alkos/libc/libssp/libssp.cpp

```diff
@@ -0,0 +1,115 @@
+// ------------------------------
+// Includes
+// ------------------------------
+
+/* main include */
+#include "libssp.hpp"
+
+/* external includes */
+#include <stdint.h>
+
+/* internal includes */
+#include "../include/defines.hpp"
+
+#ifdef __ALKOS_KERNEL__
+#include "../../kernel/include/panic.hpp"
```

Comment:
include thingy

Reply:
And this is especially horendous way of including a thing from the kernel inside libc. I don't think it should be like that at all.

Reply:
In general the current scheme with libc/libk and kernel is as such - when you compile as libc, it should generate a libc.a which is in /usr/lib/libc.a and programs can link against it - this is currently implemeneted.

libk is there to avoid code duplication - it's almost the same set of things, except you can't use syscalls in libk functions. In light of this, im not sure if libssp should even be in libk - and it raises another issue - we ought to make some guidelines and a clear disctinction as to what needs to be in kernel and what in libk and what's the difference

Reply:
Im pretty sure libssp should be placed in libc as normal application also links against it and expect libssp to be provided by libc implementation. The problem arises that kernel requires different failure handling (KernelPanic() here). Maybe KernelPanic should be inside libc?